File: javascript/027.js
/**
 * @param {number[]} nums
 * @param {number} val
 * @return {number}
 */
var removeElement = function(nums, val) {
  let last = nums.length - 1;
  for (let i = last; i >= 0; --i) {
    if (nums[i] !== val) continue;

    if (i === last) {
      --last;
    } else {
      nums[i] = nums[last];
      nums[last] = val;
      --last;
    }
  }
  
  return last + 1;
};

File: ruby/027.rb
# @param {Integer[]} nums
# @param {Integer} val
# @return {Integer}
def remove_element(nums, val)
  insert_index = 0
  0.upto(nums.length - 1).each do |i|
    if nums[i] != val
      nums[insert_index] = nums[i]
      insert_index += 1
    end
  end
  insert_index
end